### PR TITLE
Refactor optional imports

### PR DESCRIPTION
**Pull request by zduey**
This PR removes support for `BlackfynnPath` since it has sadly gone unused and only adds maintenance effort. There is a slight refactor of the code so that everything is not dumped in a single module. Optional dependencies `s3fs` and `boto3` are only imported when an `S3Path` is constructed. This allows pathman to be used for local paths without adding those non-trivial dependencies.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,7 +11,8 @@
 Pathman
 =======
 
-Pathman is a utility for interacting with files using a uniform interface regardless of where
-the files live: locally, on S3, or in some other remote service. The interface attempts closely
-follow the `pathlib` interface when possible. Support for interacting with files in S3 is done 
-through heavy use of the s3fs_ library.
+Pathman is a utility for interacting with files using a uniform interface
+regardless of where the files live: locally, on S3, or in some other remote
+service. The interface attempts closely follow the `pathlib` interface when
```
Comment (sLopezdeDiego) [inline README.rst:16]: Possible missing word:
```suggestion
service. The interface attempts to closely follow the `pathlib` interface when
```

File: docs/conf.py [hunk 2]
```diff
@@ -173,10 +133,18 @@
...
+        author,
+        "pathman",
+        "One line description of project.",
+        "Miscellaneous",
+    )
 ]
 
 
 # -- Extension configuration -------------------------------------------------
+napoleon_google_docstring = False
```
Comment (sLopezdeDiego) [inline docs/conf.py:149]: should this be set to False?

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 s3fs
 boto3
 requests
-blackfynn
\ No newline at end of file
+blackfynn
```
Comment (sLopezdeDiego) [inline requirements.txt:4]: Is `blackfynn` still a dependency here?

Comment (zduey) [reply requirements.txt:4]: I removed it right after requesting the review, so you may not have seen the change

File: docs/conf.py (hunk 2 above)
Comment (zduey) [reply docs/conf.py:149]: yes, because we are using numpy-style docstrings

File: requirements.txt (hunk 3 above)
Comment (sLopezdeDiego) [reply requirements.txt:4]: Oh, that's strange. It looked like it was just added on my end. Probably a github issue.

**Pull request merged**